e2e-tests/portal/cypress/e2e/journeys/admin/restrict-access/restrict-access.spec.js

```diff
@@ -1,29 +1,30 @@
-const deals = require('../../../../fixtures/deal-dashboard-data');
 const relative = require('../../../relativeURL');
 const { ADMIN, BANK1_MAKER1, BANK1_CHECKER1, BANK1_READ_ONLY1, BANK1_PAYMENT_REPORT_OFFICER1 } = require('../../../../../../e2e-fixtures/portal-users.fixture');
 
 context('Only allow authorised users to access admin pages', () => {
+  let dealId;
   it('should allow admins access to restricted pages', () => {
     cy.login(ADMIN);
     cy.visit('/admin/users/');
     cy.url().should('eq', relative('/admin/users/'));
   });
 
   describe('Access a deal', () => {
-    let deal;
-
     before(() => {
-      const aDealInStatus = (status) => deals.filter((aDeal) => status === aDeal.status)[0];
-
       cy.deleteDeals(ADMIN);
-      cy.insertOneDeal(aDealInStatus('Draft'), BANK1_MAKER1).then((insertedDeal) => {
-        deal = insertedDeal;
-      });
+      cy.createBssEwcsDeal();
     });
 
     it('allows read only user with all bank access to view deal', () => {
-      cy.loginGoToDealPage(ADMIN, deal);
-      cy.url().should('eq', relative(`/contract/${deal._id}`));
+      cy.loginGoToDealPage(ADMIN);
+      cy.getDealIdFromUrl(dealId).then((id) => {
+        dealId = id;
+        cy.url().then((url) => {
+          dealId = url.split('/').pop();
+          expect(dealId).to.be.a('string');
+          cy.url().should('eq', relative(`/contract/${dealId}`));
+        });
+      });
```

Comment:
why is the dealId being obtained twice?

```suggestion
      cy.getDealIdFromUrl(dealId).then((id) => {
        cy.url().should('eq', relative(`/contract/${dealId}`));
      });
```

Reply:
Done